crates/ruff/src/message/gitlab.rs

```diff
@@ -82,10 +82,16 @@ impl Serialize for SerializedMessages<'_> {
                 |project_dir| relativize_path_to(message.filename(), project_dir),
             );
 
+            let mut message_fingerprint = fingerprint(message, "");
+            while fingerprints.contains(&message_fingerprint) {
+                message_fingerprint = fingerprint(message, &message_fingerprint);
+            }
```

Comment:
A comment about the purpose of this may be useful.

Doesn't this make the fingerprint change based on previous fingerprints? Won't this dependency change a downstream fingerprint if a conflicting upstream diagnostic is fixed?

Reply:
My thinking is that "value for money", just doing this will be a lot less annoying and give a lot less noise in the gitlab reports, than having code position included, which would trigger for almost no reason. At least with this method the false reporting of fixed/new issues will either be either when a new issue is introduced, which has the same name, or an issue which there are multiples are with the same name, is fixed.

Reply:
The main "issue" I see with the heuristic is that GitLab will mark the wrong diagnostic as fixed if there's a hash collision, but I haven't developed a better heuristic. 

Let's say we have two unused variable diagnostics in a file:

```python
x = 1
y = 2
```

There's a hash collision for `x` and `y`, so `y` performs a second hashing round

and we now fix the first violation

```python
x = 1
y = 2
print(x)
```

There's no longer a hash collision for `y`, meaning that the diagnostic for `y` now gets the hash (fingerprint) of the violation that used to be for `x`. 

We could consider including more information when a hash collision occurs but not sure what information that would be that doesn't suffer from the original issue and distinguishes multiple diagnostics sufficiently.